packages/babel-core/src/config/full.js

```diff
@@ -147,6 +147,8 @@ export default gensync<[any], ResolvedConfig | null>(function* loadFullConfig(
   const opts: Object = optionDefaults;
   mergeOptions(opts, options);
 
+  if (!isCompilableFile(context.filename, opts.extensions)) return null;
```

Comment:
After https://github.com/babel/babel/pull/11907 is merged, I need to check how to make Babel report these files as `"unsupported"`.